VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionPostsThreadFragment.java

```diff
@@ -260,7 +263,10 @@ private void populatePostList() {
             getThreadListTask.cancel(true);
         }
 
-        getThreadListTask = new GetThreadListTask(getActivity(), courseData.getCourse().getId(), isAllTopics() ? null : discussionTopic.getIdentifier(),
+        getThreadListTask = new GetThreadListTask(
+                getActivity(),
+                courseData.getCourse().getId(),
+                isAllTopics() ? new ArrayList<String>() : discussionTopic.getAllTopicIds(),
```

Comment:
Can we not pass `null` to the API for the `List` parameter? If not, then we can replace the new empty `ArrayList` with `Collections.EMPTY_LIST` to avoid creating a new unused object.